### DISCUSSION TITLE
(🎁) Use pyprojectx

### DISCUSSION THREAD

**Discussion by KotlinIsland** [johnthagen/python-blueprint#219] [Ideas]
In our experience, [pyprojectx](https://pyprojectx.github.io/) is a massive upgrade for poetry etc workflows.

Additionally, pipx will not be needed anymore.

(I would also recommend ditching poetry, and adopt pdm or rye)

**Comment by johnthagen** [comment]
Thanks for the suggestion! I hadn't heard of `pyprojectx` before.

Given this project is prescriptive for others, and `pyprojectx` only has 89 stars at the moment, I wouldn't be comfortable recommending it just yet.

> (I would also recommend ditching poetry, and adopt pdm or rye)

Poetry is close to supporting PEP 621: https://github.com/python-poetry/poetry/pull/9135

I'm curious, are there any other reasons to prefer PDM over Poetry?

As for rye, I am tracking `uv` and plan to switch to it from Poetry once it has gained all of the Cargo-like capabilities that are needed. As Charlie explains, Rye is really uv/Astrals experimental playground, so I don't think I'd personally would recommend it yet broadly.

**Comment by lovepamthailand1672538** [comment]
bc1qnppvarg8f0gzcfhmytuky0l5ncc90e5kyuxl8q
